lib/pgBackRest/Storage/Helper.pm

```diff
@@ -0,0 +1,129 @@
+
+use Exporter qw(import);
+    our @EXPORT = qw();
+use File::Basename qw(basename);
+
+use pgBackRest::Common::Log;
+use pgBackRest::Config::Config;
+use pgBackRest::Storage::Posix::Driver;
+use pgBackRest::Storage::Local;
+use pgBackRest::Version;
+
+####################################################################################################################################
+# Storage constants
+####################################################################################################################################
+use constant STORAGE_LOCAL                                          => '<LOCAL>';
+    push @EXPORT, qw(STORAGE_LOCAL);
+
+use constant STORAGE_SPOOL                                          => '<SPOOL>';
+    push @EXPORT, qw(STORAGE_SPOOL);
+use constant STORAGE_SPOOL_ARCHIVE_OUT                              => '<SPOOL:ARCHIVE:OUT>';
+    push @EXPORT, qw(STORAGE_SPOOL_ARCHIVE_OUT);
+
+####################################################################################################################################
+# Compression extension
+####################################################################################################################################
+use constant COMPRESS_EXT                                           => 'gz';
+    push @EXPORT, qw(COMPRESS_EXT);
+
+####################################################################################################################################
+# Temp file extension
+####################################################################################################################################
+use constant STORAGE_TEMP_EXT                                       => BACKREST_EXE . '.tmp';
+    push @EXPORT, qw(STORAGE_TEMP_EXT);
+
+####################################################################################################################################
+# Cache storage so it can be retrieved quickly
+####################################################################################################################################
+my $hStorage;
+
+####################################################################################################################################
+# storageLocal - get local storage
+#
+# Local storage is always read-only and can never reference a remote path.  Used for adhoc activities like reading pgbackrest.conf.
+####################################################################################################################################
+sub storageLocal
+{
+    # Assign function parameters, defaults, and log debug info
+    my
+    (
+        $strOperation,
+        $strPath,
+    ) =
+        logDebugParam
+        (
+            __PACKAGE__ . '::storageLocal', \@_,
+            {name => 'strPath', default => '/', trace => true},
+        );
+
+    # Create storage if not defined
+    if (!defined($hStorage->{&STORAGE_LOCAL}{$strPath}))
+    {
+        # Create local storage
```

Comment:
Oops, actually there is one case where it is not read-only so I amended the comment.